core/Plugin/Archiver.php

```diff
@@ -83,6 +87,92 @@ public function __construct(ArchiveProcessor $processor)
+    {
+        $transientCache = Cache::getTransientCache();
+        $cacheKey = CacheId::siteAware('Archiver.RecordBuilders');
+
+        $recordBuilders = $transientCache->fetch($cacheKey);
+        if ($recordBuilders === false) {
+            $recordBuilderClasses = Manager::getInstance()->findMultipleComponents('RecordBuilders', ArchiveProcessor\RecordBuilder::class);
+
+            $recordBuilders = array_map(function ($className) {
+                if ((new \ReflectionClass($className))->getConstructor()->getNumberOfRequiredParameters() == 0) {
+                    return StaticContainer::getContainer()->make($className);
+                }
+            }, $recordBuilderClasses);
+            $recordBuilders = array_filter($recordBuilders);
+
+            /**
+             * Triggered to add new RecordBuilders that cannot be picked up automatically by the platform.
+             * If you define RecordBuilders that take a parameter, for example, an ID to an entity your plugin
+             * manages, use this event to add instances of that RecordBuilder to the global list.
+             *
+             * **Example**
+             *
+             *     public function addRecordBuilders(&$recordBuilders)
+             *     {
+             *         $recordBuilders[] = new MyParameterizedRecordBuilder($idOfThingToArchiveFor);
+             *     }
+             *
+             * @param ArchiveProcessor\RecordBuilder[] $recordBuilders An array of RecordBuilder instances
+             * @api
+             */
+            Piwik::postEvent('Archiver.addRecordBuilders', [&$recordBuilders]);
+
+            /**
+             * Triggered to filter / restrict reports.
+             *
+             * **Example**
+             *
+             *     public function filterRecordBuilders(&$recordBuilders)
+             *     {
+             *         foreach ($reports as $index => $recordBuilder) {
+             *              if ($recordBuilders instanceof AnotherPluginRecordBuilder) {
+             *                  unset($reports[$index]);
+             *              }
+             *         }
+             *     }
+             *
+             * @param ArchiveProcessor\RecordBuilder[] $recordBuilders An array of RecordBuilder instances
+             * @api
+             */
+            Piwik::postEvent('Archiver.filterRecordBuilders', [&$recordBuilders]);
+
+            $transientCache->save($cacheKey, $recordBuilders);
+        }
+
+        $requestedReports = $this->processor->getParams()->getArchiveOnlyReport();
+        if (!empty($requestedReports)) {
+            $requestedReports = is_string($requestedReports) ? [$requestedReports] : $requestedReports;
+
+            $recordBuilders = array_filter($recordBuilders, function (ArchiveProcessor\RecordBuilder $builder) use ($requestedReports) {
+                return $builder->isBuilderForAtLeastOneOf($this->processor, $requestedReports);
+            });
+        }
```

Comment:
Can we use `getArchiveOnlyReportAsArray` here?

```suggestion
        $requestedReports = $this->processor->getParams()->getArchiveOnlyReportAsArray();
        if (!empty($requestedReports)) {
            $recordBuilders = array_filter($recordBuilders, function (ArchiveProcessor\RecordBuilder $builder) use ($requestedReports) {
                return $builder->isBuilderForAtLeastOneOf($this->processor, $requestedReports);
            });
        }
```